### PR TITLE
chore(#551): adds dev setup using openshift template and few make target

### PR DESCRIPTION
**Pull request by dipak-pawar**
Adds following make target for minishift local setup for running auth,wit, ui
- **auth:**
  - `deploy-auth`
  - `deploy-auth-secrets`
  - `deploy-auth-db`
  - `deploy-auth-only`
  - `clean-auth`
  - `redeploy-auth`
 
- **wit:**
  - `deploy-wit`
  - `deploy-wit-secrets`
  - `deploy-wit-db`
  - `clean-wit`

- **f8ui**:
  - `deploy-f8ui`
  - `clean-f8ui`

- **all**:
  - `clean-all`
  - `clean-all-resources`

- **container/openshift**:
  - `build-image`
  - `expose-minishift-docker-env`
  - `init-project`

Fixes: #551

### CODE REVIEW

File: .make/Makefile.dev [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+redeploy-auth: build build-image deploy-auth-only
+
+.PHONY: minishift-start
+minishift-start:
+	minishift start --cpus 4 --memory 8GB
+	./minishift/check_hosts.sh
+	-eval `minishift docker-env` && oc login -u developer -p developer
+
+.PHONY: dev-deploy-all
+dev-deploy-all: prebuild-check deps generate build init-project clean-all-dc deploy-auth deploy-wit deploy-f8ui
```
Comment (hemanik) [inline .make/Makefile.dev]: There is no target for `clean-all-dc` probably you meant `clean-dev-all`.

File: minishift/dev_setup.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+You can get f8ui route by using `oc get routes`. It should be in format `f8ui-fabric8-services.${minishift ip}.nip.io`
+You can try logging by hitting this in browser `http://f8ui-fabric8-services.${minishift ip}.nip.io`(e.g. `http://f8ui-fabric8-services.192.168.42.177.nip.io`).
+
+Note: However if you are trying this first time you should approve your username from keycloak, so that you will be authenticated user.
+Also make sure to whitelist the domain which you are using for auth to work it as per expectation.
+
+#### Deploying Auth, WIT, UI together
+To deploy `auth`, `wit`, `fabric8-ui` together we have following target:
+```bash
+make deploy-dev-all
```
Comment (hemanik) [inline minishift/dev_setup.md]: Incorrect target name `deploy-dev-all` probably you meant `dev-deploy-all`.

**Comment by hemanik** [comment]
@dipak-pawar, pods are not getting created correctly for me. Output and logs as follows. 

```
$ oc get pods
NAME              READY     STATUS             RESTARTS   AGE
auth-1-2tjpl      0/1       ImagePullBackOff   0          2m
auth-1-deploy     1/1       Running            0          3m
db-auth-1-d6glv   1/1       Running            0          3m
db-wit-1-8lg92    0/1       Pending            0          2m
db-wit-1-deploy   1/1       Running            0          3m
f8ui-1-95k8h      1/1       Running            0          2m
wit-1-5757c       0/1       CrashLoopBackOff   5          2m
wit-1-deploy      1/1       Running            0          2m
```  

```
 $ oc logs auth-1-2tjpl
Error from server (BadRequest): container "auth" in pod "auth-1-2tjpl" is waiting to start: trying and failing to pull image
```

```
$ oc logs wit-1-5757c
pmcd: cannot open log "/dev/force-logging-to-stderr" for writing : Permission denied
Log for pmcd on wit-1-5757c started Wed Jul 18 19:12:08 2018


active agent dom   pid  in out ver protocol parameters
============ === ===== === === === ======== ==========
pmcd           2                 2 dso i:6  lib=/var/lib/pcp/pmdas/pmcd/pmda_pmcd.so entry=pmcd_init [0x7f21892f7500]
proc           3                 2 dso i:7  lib=/var/lib/pcp/pmdas/proc/pmda_proc.so entry=proc_init [0x7f21890d6620]
linux         60                 2 dso i:7  lib=/var/lib/pcp/pmdas/linux/pmda_linux.so entry=linux_init [0x7f2188ea2d90]
prometheus   144   120  13  14   2 bin pipe cmd=python /var/lib/pcp/pmdas/prometheus/pmdaprometheus.python

Host access list empty: host-based access control turned off
User access list empty: user-based access control turned off
Group access list empty: group-based access control turned off


pmcd: PID = 6, PDU version = 2
pmcd request port(s):
  sts fd   port  family address
  === ==== ===== ====== =======
  ok     5       unix   /var/run/pcp/pmcd.socket
  ok     3 44321 inet   INADDR_ANY
  ok     4 44321 ipv6   INADDR_ANY
[Wed Jul 18 19:12:09] pmcd(6) Info: PMNS file "DEFAULT" is unchanged
time="2018-07-18T19:12:12Z" level=warning msg="unable to parse log level configuration error: \"not a valid logrus Level: \\\"\\\"\"" 
time="2018-07-18 19:12:13" level=warning msg="failed to get current user" err="user: unknown userid 1000130000" file="/tmp/go/src/github.com/fabric8-services/fabric8-wit/main.go" func=main.printUserInfo pkg=main 
time="2018-07-18 19:12:18" level=error msg="ERROR: Unable to open connection to database dial tcp 172.30.28.197:5432: i/o timeout" 
time="2018-07-18 19:12:18" level=info msg="Retrying to connect in 1s..." 
```
Further, the databases are missing all tables probably due to the services not getting deployed correctly.

**Comment by dipak-pawar** [comment]
@hemanik, From the pod logs, I can see auth pod is not ready it's getting `ImagePullBackOff`. Can you make sure that you have `auth` image in `minishift` while retrying it.

Also I can see `db-wit` pod is in a pending state due to insufficient space on node. I hope you started minishift with `--memory 8` as defined in `minishift start` goal.

**Comment by alien-ike** [comment]
Hey @hemanik! It seems you tried to trigger `/ok-without-tests` command, but this will not have any effect due to insufficient permission. You have to be admin or requested reviewer or pull request approver, but not pull request creator for this command to take an effect.

**Comment by hemanik** [comment]
Thanks @dipak-pawar. LGTM :+1:

**Comment by sbose78** [comment]
@dipak-pawar and @hemanik , thank you for this - could one of your please update PR description with the new set of commands supported ? I'm gonna use the same in the commit message.

**Comment by dipak-pawar** [comment]
@sbose78 I have updated the PR description. If it looks good to you go ahead with merge

**Comment by sbryzak** [comment]
Very nice new make targets, and I appreciate the good documentation!

**Pull request merged**